### PR TITLE
feat: submitting private link and be displayed real-time

### PR DESCRIPTION
**Pull request by sshanzel**
We now accept parameters for the title and image to use for the external link, but to support backwards compatibility, the said parameters are kept optional.

### CODE REVIEW

File: src/entity/posts/utils.ts [hunk 1]
```diff
@@ -378,6 +385,8 @@ export const createExternalLink = async (
       sourceId: UNKNOWN_SOURCE,
       url,
       canonicalUrl: url,
+      title,
+      image,
       sentAnalyticsReport: true,
       private: true,
       origin: PostOrigin.Squad,
```
Comment (rebelchris) [inline src/entity/posts/utils.ts:392]: Should we not do something with this visible as well? (Line below)

File: src/schema/posts.ts [hunk 2]
```diff
@@ -880,16 +894,12 @@ export const resolvers: IResolvers<any, Context> = {
...
-        sourceId,
-        url,
-        commentary,
-      }: { sourceId: string; url: string; commentary: string },
+      { sourceId, commentary, url, title, image }: SubmitExternalLinkArgs,
       ctx,
     ): Promise<GQLEmptyResponse> => {
-      await ctx.con.transaction(async (manager) => {
-        await ensureSourcePermissions(ctx, sourceId, SourcePermissions.Post);
+      await ensureSourcePermissions(ctx, sourceId, SourcePermissions.Post);
```
Comment (rebelchris) [inline src/schema/posts.ts:900]: Why did you move this outside of the transaction?

File: src/entity/posts/utils.ts (hunk 1 above)
Comment (sshanzel) [reply src/entity/posts/utils.ts:392]: This is the article with unknown source which should still always be `visible: false`.

**Comment by sshanzel** [comment]
> Also are you going to create the actual call in a separate PR?

Yes. Figured it would be difficult knowing the current setup, so decided to isolate it.

File: src/schema/posts.ts (hunk 2 above)
Comment (sshanzel) [reply src/schema/posts.ts:900]: Oh, was working on something, and ended up removing it. This got left. Though this is not utilized/needed in the transaction so nothing should be different performance-wise. Probably even better since we don't have to initialize the transaction while it is not yet needed.

File: src/entity/posts/utils.ts (hunk 1 above)
Comment (rebelchris) [reply src/entity/posts/utils.ts:392]: That's not really correct, they also become visible when the title is available :)
They stay private at all times is what you mean.

This case would cause the error of it showing the share post, but not being able to load the actual title.

File: src/schema/posts.ts (hunk 2 above)
Comment (rebelchris) [reply src/schema/posts.ts:900]: Does query, so would prefer it to stay in the transaction to not loose scope.

File: src/entity/posts/utils.ts (hunk 1 above)
Comment (sshanzel) [reply src/entity/posts/utils.ts:392]: Oh, in that case, my understanding was wrong. Thanks for pointing it out, let me update it.

File: src/schema/posts.ts (hunk 2 above)
Comment (sshanzel) [reply src/schema/posts.ts:900]: Oh, cool. Let's keep it inside.

Comment (rebelchris) [reply src/schema/posts.ts:900]: It's still outside 👀

**Review by rebelchris — CHANGES_REQUESTED** [review]

File: src/schema/posts.ts (hunk 2 above)
Comment (sshanzel) [reply src/schema/posts.ts:900]: Too many PRs going left and right. Forgive me this 😭

**Review by rebelchris — APPROVED** [review]
Nice! 🙏

**Comment by sshanzel** [comment]
@capJavert would love to hear what you think 🚢 🚢 🚢

**Review by capJavert — APPROVED** [review]
Nice and simple 🚀
